com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/VertexAnimation.template.hlsl

```diff
@@ -39,7 +39,15 @@ VertexDescriptionInputs AttributesMeshToVertexDescriptionInputs(AttributesMesh i
     return output;
 }
 
-AttributesMesh ApplyMeshModification(AttributesMesh input, float3 timeParameters)
+// This is used for injecting the define below.
+$splice(CustomInterpolatorPreVertex)
+
+#if defined(USE_CUSTOMINTERP_APPLYMESHMOD) // mirrored in VertMesh.hlsl
+// use ifdef via TESSELLATION_ON to use VaryingsMeshToDS (Domain varyings instead of pixel varyings) whenever SG is modified to support Tess.
```

Comment:
is this coment a TODO? For the future support of tessellation?

Reply:
I have a suggestion. To avoid too introduce a new function, you could do 

AttributesMesh ApplyMeshModification(AttributesMesh input, float3 timeParameters
#if defined(USE_CUSTOMINTERP_APPLYMESHMOD)
, inout VaryingsMeshToPS varyings
#endif
)

This would be a bit cleaner and less code duplication. Same in vertmesh and motion vector.  Waht do you think? thanks

Reply:
Yes- it's a todo. The work to get Custom Interpolators supporting Tess, given current code, should be minimal, but I didn't want to pump that in until that is actually a feature that both HDRP and SG are good to go on.
--
I opted for a verbose/explicit ifdef, but yea, I don't have an issue with making that more concise.